packages/flutter/test/widgets/editable_text_show_on_screen_test.dart

```diff
@@ -489,6 +491,242 @@ void main() {
+        final ScrollController editableScrollController = ScrollController();
+
+        await tester.pumpWidget(
+          buildEditableText(
+            rejectUserInputs: false,
+            scrollController: scrollController,
+            editableScrollController: editableScrollController,
+          ),
+        );
+
+        focusNode.requestFocus();
+        await tester.pumpAndSettle();
+
+        // Ensure the caret is not fully visible and the text field is focused.
+        scrollController.jumpTo(0);
+        editableScrollController.jumpTo(0);
+        await tester.pumpAndSettle();
+        expect(isCaretOnScreen(tester), isFalse);
+
+        final EditableTextState state = tester.state<EditableTextState>(
+          find.byType(EditableText, skipOffstage: false),
+        );
+
+        // Change the selection. Show caret on screen even when readyOnly is
+        // false.
+        state.textEditingValue = state.textEditingValue.copyWith(selection: const TextSelection.collapsed(offset: 90));
+        await tester.pumpAndSettle();
+        expect(isCaretOnScreen(tester), isTrue);
+        expect(scrollController.offset, greaterThan(0.0));
+        expect(editableScrollController.offset, greaterThan(0.0));
+
+        // Rejects user input.
+        await tester.pumpWidget(
+          buildEditableText(
+            rejectUserInputs: true,
+            scrollController: scrollController,
+            editableScrollController: editableScrollController,
+          ),
+        );
+
+        // Ensure the caret is not fully visible and the text field is focused.
+        scrollController.jumpTo(0);
+        editableScrollController.jumpTo(0);
+        await tester.pumpAndSettle();
+        expect(isCaretOnScreen(tester), isFalse);
+
+        state.textEditingValue = state.textEditingValue.copyWith(selection: const TextSelection.collapsed(offset: 100));
+        await tester.pumpAndSettle();
+        expect(isCaretOnScreen(tester), isTrue);
+        expect(scrollController.offset, greaterThan(0.0));
+        expect(editableScrollController.offset, greaterThan(0.0));
+      });
+
+      // Regression text for https://github.com/flutter/flutter/pull/74722.
+      testWidgets('does NOT randomly trigger when cursor blinks', (WidgetTester tester) async {
+        textEditingController.text = 'a' * 100;
+        textEditingController.selection = const TextSelection.collapsed(offset: 1);
+        final ScrollController scrollController = ScrollController(initialScrollOffset: 100);
+        final ScrollController editableScrollController = ScrollController();
+        final bool deterministicCursor = EditableText.debugDeterministicCursor;
+        EditableText.debugDeterministicCursor = false;
+
+        await tester.pumpWidget(
+          buildEditableText(
+            rejectUserInputs: false,
+            scrollController: scrollController,
+            editableScrollController: editableScrollController,
+          ),
+        );
+
```

Comment:
I run this test on the master branch but it passed. It should be failed, right?

It seems like here miss a `await tester.pumpAndSettle();`

Reply:
This is the first build and there's no animation. I don't think we need `pumpAndSettle` here.

Reply:
Oh I see what you are saying.

Reply:
hah had to remove the `ListView`. Not this should fail on `master`. Thanks for catching this!

Reply:
Yes, it works.
I also wonder why the list will scroll a little on the first frame? Is it a ballistic scroll?

Reply:
the `ListView` didn't seem to have scrolled (scrolloffset didn't change), but the resulting `caretRect` has a slightly different `top` value (before appending an "a" : `0.0` -> after appending an "a": `-0.0`). Not sure exactly why but could be accumulated rounding errors.

Reply:
I see, the `ScrollController.initialScrollOffset` of 100 pixels has overscrolled.